testing/kokoro/version.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Determine if the given module should be tested in the current environment.
+//
+// If versions cannot be evaluated, we fail "successful" to ensure tests are
+// run by default.
+//
+// Usage:
+// go run . -module=../../run/hello-broken
+// This will exit with a non-zero status code if the module has a higher
+// go version than the runtime.
+
+// Command version checks for a module vs. environment go version mismatch
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+)
+
+// maxVersion is the most advanced go version in testing.
+const maxVersionStr="1.13"
```

Comment:
Should probably document somewhere this needs to be updated. Maybe we can auto-detect from listing the `*.cfg` files in this directory?

Reply:
Maybe we could have a metadata.go file at the repo root which defines some constants like this? I'd rather not have testing code dependent on the CI system config.